server/src/main/java/org/elasticsearch/action/admin/indices/shards/IndicesShardStoresResponse.java

```diff
@@ -213,21 +211,16 @@ public int compareTo(StoreStatus other) {
      * Single node failure while retrieving shard store information
      */
     public static class Failure extends DefaultShardOperationFailedException {
-        private String nodeId;
+        private final String nodeId;
 
         public Failure(String nodeId, String index, int shardId, Throwable reason) {
             super(index, shardId, reason);
             this.nodeId = nodeId;
         }
 
         private Failure(StreamInput in) throws IOException {
-            if (in.getVersion().before(Version.V_7_4_0)) {
```

Comment:
Since this PR is labeled to be backported to 7.10, shouldn't we keep this as is so the backport is easy, and cleanup in a separate change just for master?

Reply:
Makes sense, less noisy I'll batch this with a few other cleanups of outdated version handling in master soon :)